fs/copy_linux_test.go

```diff
@@ -27,10 +28,87 @@ import (
 	"path/filepath"
 	"testing"
 
+	"github.com/containerd/continuity/sysx"
 	"github.com/containerd/continuity/testutil"
 	"github.com/containerd/continuity/testutil/loopback"
 )
 
+func TestCopyDirWithXAttrExcludes(t *testing.T) {
+	src, err := ioutil.TempDir("", "containerd-test-copy-dir-source-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(src)
+
+	if err := sysx.Setxattr(src, "user.test-1", []byte("one"), 0); err != nil {
+		t.Fatal(err)
+	}
+	if err := sysx.Setxattr(src, "user.test-2", []byte("two"), 0); err != nil {
+		t.Fatal(err)
+	}
+	if err := sysx.Setxattr(src, "user.test-x", []byte("three-four-five"), 0); err != nil {
+		t.Fatal(err)
+	}
+
+	t.Run("none", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", "containerd-test-copy-dir-with-xattr-")
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer os.RemoveAll(dir)
+		err = CopyDir(dir, src)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if one, err := sysx.Getxattr(dir, "user.test-1"); err != nil {
+			t.Fatal(err)
+		} else if string(one) != "one" {
```

Comment:
Can we get rid of the `else if` clasues?

`t.Fatal` will take care of it.

Reply:
Sure. I tend to prefer the limited scope of these kind of if-then-else blocks and the implied lack of redefinition and/or var bleed-through, keeps things tidy. I can work up something a little more mainstream I guess.

Reply:
@cpuguy83 I bit the bullet and adopted the local **fs/fstest** package and moved these changes into fs/copy_unix_test.go as well as implemented a generic-enough-for-our-testing-purposes **assertXAttr** func which uses a switch instead of if-else